### ISSUE TITLE
Wrong address used in call_soap_method_xml

### ISSUE THREAD

**Issue by ssallmen-pro** [Altran-PT-GDC/Robot-Framework-SOAP-Library#34]
I think that using self.url as the destination address in `call_soap_method_xml` when zeep's `client.transport.post_xml` is called is not right. The service binding address should be used instead. Using `self.url` as the address for `client.transport.post_xml` will make the library impossible to be used when the WSDL address is not the same as the service address.

I have a case like this. In url `http://myservicehost.com/webtools/control/WebService/getContactPersons?wsdl` I have a WSDL that has binding and service defined as follows:
```xml
<wsdl:binding name="GetContactPersonsSoapBinding" type="tns:GetContactPersonsPortType">
<soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>
<wsdl:operation name="getContactPersons">
<soap:operation soapAction="http://myservicehost.com/webtools/control/WebService" style="document"/>
<wsdl:input name="input">
<soap:body parts="getContactPersons" use="literal"/>
</wsdl:input>
<wsdl:output name="output">
<soap:body parts="getContactPersonsResponse" use="literal"/>
</wsdl:output>
<wsdl:fault name="fault">
<soap:fault name="fault" use="literal"/>
</wsdl:fault>
</wsdl:operation>
</wsdl:binding>
<wsdl:service name="getContactPersons">
<wsdl:port binding="tns:GetContactPersonsSoapBinding" name="getContactPersonsPort">
<soap:address location="http://myservicehost.com/webtools/control/WebService"/>
</wsdl:port>
</wsdl:service>
```
Now, when I try to use `Call SOAP Method XML` keyword to send my getContactPersons operation message, it will post the message to the same url from where the WSDL was loaded (`http://myservicehost.com/webtools/control/WebService/getContactPersons?wsdl`), and not to the service address (`http://myservicehost.com/webtools/control/WebService`) as it should.

I updated the call to `self.client.transport.post_xml` in `call_soap_method_xml` as follows, and it worked, but I don't know if that is the right fix for the issue as I don't know enough about the zeep library.
```python
        response = self.client.transport.post_xml(address=self.client.service._binding_options['address'], envelope=xml_obj, headers=headers)
```
Here is the full WSDL: [getContactPersons.wsdl.txt](https://github.com/Altran-PT-GDC/Robot-Framework-SOAP-Library/files/8799541/getContactPersons.wsdl.txt)

And here is my request xml-file: [getContactPersons.xml.txt](https://github.com/Altran-PT-GDC/Robot-Framework-SOAP-Library/files/8799542/getContactPersons.xml.txt)

Files were modified for confidential information and added txt as the file extension as GitHub seems not to support attaching wsdl or xml files.

**Comment by ssallmen-pro** [comment]
Just for additional information here is Robot test case:
```
*** Settings ***
Library    SoapLibrary

*** Variables ***
@{my creds}    user    password
*** Test Cases ***
Get Contacts
    Create Soap Client    http://myservicehost.com/webtools/control/WebService/getClaudiaContactPersons?wsdl    auth=${my creds}
    # ${response}=    Call SOAP Method    getClaudiaContactPersons    contactId    0035E00001gxdx4QAA
    ${response}=    Call SOAP Method With XML    ${CURDIR}/../data/getAllClaudiaContactPersons.xml
    Save XML To File    ${response}    ${CURDIR}    contacts
```

And here you can see the output telling where the requests were sent. The contacts.xml will contain the response, which is the WSDL because the SOAP request was sent to the same address where the WSDL is located.
![robot_log_html](https://user-images.githubusercontent.com/98014506/171031097-2a6dc32f-ed13-462a-bb95-9dd5b28eb542.png)

I think the same issue has been reported already way back (#20) and it was said to be fixed with some irrelevant fix but I don't think that fix was really of help at all. It seems that service has exactly the same issue as there is the service address definition in the mentioned WSDL, different to location of the WSDL.

**Comment by samuelpcabral** [comment]
Hi @ssallmen-pro Thanks for the detailed explanation and the fix suggestion. As I don't have an endpoint with the mentioned characteristics, I added your solution and ensured that the library continues to work as expected on the endpoints that I have access...

Please update with: pip install --upgrade robotframework-soaplibrary==0.9

**Comment by ssallmen-pro** [comment]
@samuelpcabral Thank you very much for your help! The new version works for me well.

**Comment by jckspencer** [comment]
Following up on this because I require the opposite solution or rather the old functionality...

I'm testing a service with internal gateway endpoints to reach the outside service. The initial WSDL comes from an internal endpoint but the binding is defined with the external endpoint. However, using that endpoint makes the request fail. 

In short, I need to send a SOAP request to the same address as my WSDL but this functionality was removed. Perhaps a compromise to serve either scenario is possible?

**Comment by ssallmen-pro** [comment]
Hmmm...why is the service binding address not working for you @jckspencer? Has the service moved and the binding addresses not updated, or do you have some restrictions in your test environment?
I would not change the tool because of some wrongly configured service binding address in WSDL, or because there is some issue with the test environment. The service binding address of a WSDL is really what should be followed, if that is available, not the location of the WSDL itself.

**Comment by jckspencer** [comment]
> Hmmm...why is the service binding address not working for you @jckspencer? Has the service moved and the binding addresses not updated, or do you have some restrictions in your test environment?
> I would not change the tool because of some wrongly configured service binding address in WSDL, or because there is some issue with the test environment. The service binding address of a WSDL is really what should be followed, if that is available, not the location of the WSDL itself.

We are proxying traffic through a [Kong gateway](https://docs.konghq.com/gateway/latest/), so the address I need to send traffic and make requests to is different from that in service's WSDL. The service is configured properly AFAIK (I wasn't involved with developing the gateway, just testing) - the service works flawlessly in SoapUI and in an older test environment that doesn't use Robot. It seems the default in SoapUI is to send the request to the same address as the WSDL if I'm not mistaken.

Like @samuelpcabral, I've not developed an API but from a testing standpoint, having the option to choose which address to use can't hurt. That said, for my pull request #40 I chose to make the service binding address the default... not sure if that is best practice but based on this post I assumed it was. Happy to discuss more as needed...

**Comment by samuelpcabral** [comment]
Hi @ssallmen-pro the new version 1.1 is released, please run: pip install --upgrade robotframework-soaplibrary

Note that I've set the old behavior as the default. then to run the keyword using binding address, you will need to pass this as argument... 

like this:

```
${response}=    Call SOAP Method With XML    ${requests_dir}${/}Request.xml    use_binding_address=True
${response}=    Call SOAP Method With String XML    ${requests_dir}${/}Request.xml    use_binding_address=True
```

**Comment by ssallmen-pro** [comment]
No, I will not update to use the new version as the old one works better for me. I don't want to update all the keyword calls to have that additional option. And, if I would update the library, I would need to update a couple of Docker images as well.
I could have done that if it would have been only in the `Create Soap Client` keyword, but not if the option is for every single request.
I will find some better library combination to call the SOAP APIs.

**Comment by ssallmen-pro** [comment]
Need to think if I will make a fork and publish that fork with a different name to PyPI so that I can continue developing that further, if I need to continue working with SOAP API testing, which I hope I don't have to do in the future.

**Comment by samuelpcabral** [comment]
@ssallmen-pro Hi I changed to pass the use_binding_address=True to `Create Soap Client` 
you had already mentioned it in the previous comment but it went completely unnoticed for me, and it really gets much better. Sorry.

I already created the version 1.2 `pip install --upgrade robotframework-soaplibrary`

Now just need to pass:
`Create Soap Client    ${wsdl}    use_binding_address=True`

If you found some issue, please let me know... Thank You.
